### PR TITLE
Add info about web servers to Rails docs

### PR DESCRIPTION
**Pull request by alexmuller**
Add info about web servers to Rails docs because this wasn't immediately obvious to me.

Still working out what kind of shape you want your documentation to be - happy to put this in GOV.UK's docs instead.

### CODE REVIEW

File: docs/deploying_apps/deploying_rails.md [hunk 1]
```diff
@@ -60,6 +60,17 @@ You can add another instance of your app by running:
...
 
+## Web servers
+
+By default, the Cloud Foundry buildpack for Rails 4 [runs `bin/rails server`][cf-buildpack-rails] to
+spin up the application, which will use WEBrick as a server.
+
+You may want to use a different server in production. The Cloud Foundry docs have
+[more information on configuring a production server][cf-docs-prodserver]
+
+[cf-buildpack-rails]: https://github.com/cloudfoundry/ruby-buildpack/blob/1f0ac3ce10866390d161c3f27e71d64890859454/lib/language_pack/rails4.rb#L27
```
Comment (mtekel) [inline docs/deploying_apps/deploying_rails.md:76]: will this link still be valid when we update the buildpack? would it need maintenance? If so, could it be replaced with something that's more useful/relevant for the future?

**Comment by bleach** [comment]
Hi @alexmuller, this is really useful stuff, thanks.

This page was intended to get a basic rails app up and running on the PaaS and we link to [the Cloud Foundry rails deployment docs](http://docs.cloudfoundry.org/buildpacks/ruby/gsg-ror.html) for a more involved guide. That page links to the production servers guide. However, judging by yours and others' experiences people don't find that documentation.

I'm inclined to keep this page focussed on "I want to deploy a toy app on CF to see how CF works" and not go into a full prod setup in it. Perhaps we should have another page or section for the use cases of "I have an existing rails app and want to move it to CF" and "I've done a toy app, now I want to put production on CF". Other things that people get caught by are doing database migrations and running background jobs.

What do you think? Paging @benh-gds who's done a lot of the recent work on our docs for his thoughts.

File: docs/deploying_apps/deploying_rails.md (hunk 1 above)
Comment (bleach) [inline docs/deploying_apps/deploying_rails.md:76]: As of Rails 5, the default is puma: http://guides.rubyonrails.org/getting_started.html#starting-up-the-web-server; maybe we could update the wording to reflect that, as I don't think this changes often enough to worry about as a maintenance issue.

I suspect most people don't want to be relying on the default, regardless of what it is at this point in time.

**Comment by benh-gds** [comment]
I think it makes sense to include this warning, and then link to the Cloud Foundry docs, so that people are made aware that the default web server isn't suitable for production. We should document all the "gotchas" - even if the answers are links to Cloud Foundry. Since we don't support everything in CF, I think we should assume our docs are the primary reference, and we will direct people to the official CF apps as appropriate. Agree that we should probably have separate "deploy a toy app" (with code of an example app and code showing how to connect to postgres),"migrate an existing app", "deploy a production app" pages for each language.

**Comment by bleach** [comment]
#17 adds a production checklist, which we think is a better place to add this information. It is not yet merged, so let's revisit after it's been reviewed.

**Comment by bleach** [comment]
In the end I edited for rails 5+ but didn't move into the production checklist docs, because on reflection I think that choosing the right webserver isn't only a consideration for production apps.

**Pull request merged**